packages/core/admin/server/policies/hasLicenseLimitAllowance.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const utils = require('@strapi/utils');
+const { ApplicationError } = require('@strapi/utils/lib/errors');
+
+const { PolicyError } = utils.errors;
+
+module.exports = async (policyCtx, config = {}) => {
+  if (!strapi.EE) return true;
+
+  const permittedSeats = strapi.ee.licenseInfo.seats;
+  if (!permittedSeats) return true;
+
+  const userCount = await strapi.db.query('admin::user').count({
+    where: { isActive: true },
+  });
+
+  if (userCount < permittedSeats) return true;
+  if (userCount >= permittedSeats && config.isCreating) {
+    throw new PolicyError("License seat limit reached, can't create new user", {
+      policy: 'license-limit-allowance',
+    });
+  }
+
+  const user = await strapi.db.query('admin::user').findOne({
+    where: { id: policyCtx.params.id },
+  });
+
+  if (!user) {
+    throw new ApplicationError('User could not be found');
```

Comment:
I guess this would happen in the scenario of updating a user that doesn't exist. But then, shouldn't we let the application layer handle this validation instead? I feel like we're giving this policy work outside of its scope.

Instead, if the user doesn't exist, why don't we just say "ok, it won't have an impact anyway, let's let the application layer decide what to do with this"

Reply:
This bit was needed for the code below it where we check if what's changed on the user is `isActive` property. User info is allowed to be changed, but not the `isActive` state. But I guess I could just use the optional chaining below and get rid of this condition.